src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -0,0 +1,260 @@
+    ResourceId resource_id = 1;
+    string display_name = 2;
+    optional string state = 3;
+    optional google.protobuf.Timestamp created_at = 4;
+    repeated EnvironmentVariable environment = 5;
+    optional int32 expected_endpoints_count = 6;
+    repeated Endpoint endpoints = 7;
+    repeated ResourceCommandRequest commands = 8;
+
+    // TODO potentially remove endpoints/env vars/anything that might not be general purpose
+
+    // List of additional data, as name/value pairs.
+    // For:
+    // - Containers: image, container_id, ports
+    // - Executables: process_id, executable_path, working_directory, arguments
+    // - Projects: process_id, project_path
+    repeated AdditionalData additional_data = 9;
+}
+
+////////////////////////////////////////////
+
+// Models a snapshot of resource state
+message WatchResourcesSnapshot {
+    repeated ResourceSnapshot resources = 1;
+    repeated ResourceType types = 2;
+}
+
+////////////////////////////////////////////
+
+message ResourceDeletion {
+    ResourceId resource_id = 1;
+}
+
+message WatchResourcesChange  {
+    oneof kind {
+        ResourceDeletion delete = 1;
+        ResourceSnapshot upsert = 2;
+    }
+}
+
+message WatchResourcesChanges {
+    repeated WatchResourcesChange value = 1;
+}
+
+////////////////////////////////////////////
+
+// Initiates a subscription for data about resources.
+message WatchResourcesRequest {
+    // True if the client is establishing this connection because a prior one closed unexpectedly.
+    optional bool is_reconnect = 1;
+}
+
+// A message received from the server when watching resources. Has multiple types of payload.
+message WatchResourcesUpdate {
+    oneof kind {
+        // Snapshot of current resource state. Received once upon connection, before any "changes".
+        WatchResourcesSnapshot initial_snapshot = 1;
+        // One or more deltas to apply.
+        WatchResourcesChanges changes = 2;
+    }
+}
+
+////////////////////////////////////////////
+
+message ConsoleLogLine {
+    string text = 1;
+    optional bool is_error = 2;
+}
+
+// Initiates a subscription for the logs of a resource.
+message WatchResourceConsoleLogsRequest {
+    // Specifies the resource to watch logs from.
+    ResourceId resource_id = 1;
+}
+
+// A message received from the server when watching resource logs.
+// Contains potentially many lines to be appended to the log.
+message WatchResourceConsoleLogsUpdate {
+    repeated ConsoleLogLine log_lines = 1;
+}
+
+////////////////////////////////////////////
+
+message StructuredLog {
+    google.protobuf.Timestamp timestamp = 1;
+    uint32 flags = 2;
+    StructuredLogLevel level = 3;
+    string application_name = 4;
+    string message = 5;
+    string span_id = 6;
+    string trace_id = 7;
+    optional string original_format = 8;
+    optional string scope_name = 9;
+}
+
+// Maps to Microsoft.Extensions.Logging.LogLevel (in names, not numbers)
+enum StructuredLogLevel {
```

Comment:
This is inconsistent with the OpenTelemetry log data model https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber and also a slippery slope, because I have seen libraries define the log levels "the other way around", namely, using lower numbers for higher severity errors. A good example is Event Tracing for Windows (ETW) https://learn.microsoft.com/en-us/windows/win32/api/evntrace/nf-evntrace-enabletraceex2

Reply:
What do you suggest?

Reply:
I would follow the OTEL log data model and be prepared that the resource service might need to do some "translation" of log levels depending on the source. 

This also is related to the discussion above https://github.com/dotnet/aspire/pull/1274/files#r1422518059 , namely the question what is the telemetry format we standardize on for the sake of populating the dashboard. Following OTEL conventions makes sense to me and makes it easier to handle differences such as 
- resource service being the store of telemetry (local dev environment) vs just exposer/transmitter of telemetry (some cloud environments, presumably)
- application services implemented in .NET vs application services implemented in other languages

Reply:
We ended up removing these types.